### ISSUE TITLE
[Feature Request] Default creation of LazyStackedTensorDict vs TensorDict when calling torch.stack?

### ISSUE THREAD

**Issue by vmoens** [facebookresearch/rl#5]
By default, `torch.stack(list_of_tds, dim)` will return a `LazyStackedTensorDict` object.
Keeping the discussion about the usefulness of this class apart, it may be more desirable to have a `LazyStackedTensorDict` created only when it is explicitely asked by the user, e.g.
```python
stacked_td = torch.stack(list_of_tds, dim)
assert isinstance(stacked_td, TensorDict) # passes

stacked_td = torch.stack(list_of_tds, dim, contiguous=False)
assert isinstance(stacked_td, LazyStackedTensorDict) # passes
```
The main advantage is that working with `LazyStackedTensorDict` can be computationally expensive and users may not be aware of this bottleneck.
Also, when called on lists of `torch.Tensor` objects, `torch.stack` does not return a non-contiguous object. Aligning `TensorDict` behaviour to this might be beneficial for consistency purposes.

**Comment by vmoens** [comment]
Closing as #352 made it clear that this is needed. Happy to re open if needed
